Core.Collectors/Collector/CachingCollector.cs

```diff
@@ -2,13 +2,32 @@
 // Licensed under the MIT License.
 
 using Microsoft.CloudMine.Core.Collectors.Cache;
+using Microsoft.CloudMine.Core.Collectors.Utility;
 using Microsoft.CloudMine.Core.Telemetry;
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace Microsoft.CloudMine.Core.Collectors.Collector
 {
+    public static class CachingCollectorUtils
+    {
+        public const int CacheLookupMultiplier = 2;
+
+        internal static bool ShallIgnoreCache(bool ignoreCache, bool scheduledCollection, DateTime utcNow, DateTime sliceEndDateUtc, TimeSpan sliceCollectionFrequency)
```

Comment:
I am not sure how an argument can be names utcNow. UtcNow is a constant value. This is very confusing for me.